### PR TITLE
[MIG][14.0] l10n_fr_siret_lookup

### PR DESCRIPTION
**Pull request by alexis-via**
Add support for the "create" scenario via onchange
Don't add custom fields on res.partner
Add VIES lookup to check that the computed VAT number is ok
This module new depends on l10n_fr_siret
Filter out closed entities in opendatasoft lookup

This PR is the consequence of the discussion that took place on https://github.com/OCA/l10n-france/pull/173 and on https://github.com/OCA/l10n-france/pull/310

### CODE REVIEW

**Comment by alexis-via** [comment]
Here is an example of the autocomplete feature on the name field (works with SIREN, SIRET and VAT). It also works on the VAT number field, SIREN and SIRET fields.

siren_lookup

**Comment by alexis-via** [comment]
I hate to rename modules, but I prefer to discuss it now before we merge this module in OCA for the first time: should we rename it to l10n_fr_siret_lookup ? As it now depend on l10n_fr_siret, maybe it would make more sense...

**Comment by alexis-via** [comment]
@remi-filament What do you think of the changes I introduced? Are you ok to rename the module to l10n_fr_siret_lookup ?

**Review by remi-filament — CHANGES_REQUESTED** [review]
Hi Alexis, thanks for bringing this to v14.0 !

First, I face an issue with your code, and more specifically with the check_vies() command since you added a timeout=5 but this timeout parameter was only made available in stdnum v1.9 (from [commit on 2018-02-06](https://arthurdejong.org/git/python-stdnum/commit/?id=5fd1ae0437da91b4642f3bdb00c175a511fa8a05)) when Odoo v14 use python-stdnum version 1.8 (from [requirements.txt](https://github.com/odoo/odoo/blob/14.0/requirements.txt))

I have then an error as follows :
```ERROR odoo14 odoo.addons.l10n_fr_siren_lookup.models.res_partner: VIES query failed: check_vies() got an unexpected keyword argument 'timeout'```

Regarding the naming, you can change the name without problem for me.
Also, you may want to remove CREDITS.rst and my name from maintainers in __manifest__.py

Thanks !

File: l10n_fr_siren_lookup/models/res_partner.py
Comment (remi-filament) [inline l10n_fr_siren_lookup/models/res_partner.py]:
```suggestion
            vies_res = check_vies(vat)
```
Timeout is not supported in check_vies for python-stdnum version 1.8 used by Odoo v14.0

File: l10n_fr_siren_lookup/i18n/fr.po
Comment (remi-filament) [inline l10n_fr_siren_lookup/i18n/fr.po]: Updated translation :
```suggestion
# Translation of Odoo Server.
# This file contains the translation of the following modules:
# 	* l10n_fr_siren_lookup
#
msgid ""
msgstr ""
"Project-Id-Version: Odoo Server 14.0\n"
"Report-Msgid-Bugs-To: \n"
"POT-Creation-Date: 2021-10-19 13:17+0000\n"
"PO-Revision-Date: 2021-10-19 13:17+0000\n"
"Last-Translator: \n"
"Language-Team: \n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: \n"
"Plural-Forms: \n"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__staff
msgid "# Staff"
msgstr "Effectifs"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__ape
msgid "APE Code"
msgstr "Code NAF"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__ape_label
msgid "APE Label"
msgstr "Libellé Code NAF"

#. module: l10n_fr_siren_lookup
#: model:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
msgid "Address"
msgstr "Adresse"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__category
msgid "Category"
msgstr "Catégorie"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__city
msgid "City"
msgstr "Ville"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_view_form
msgid "Close"
msgstr "Fermer"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
msgid "Company"
msgstr "Société"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
msgid "Company Infos"
msgstr "Infos Entreprise"

#. module: l10n_fr_siren_lookup
#: model:ir.model,name:l10n_fr_siren_lookup.model_siren_wizard_line
msgid "Company Selection"
msgstr "Sélection Entreprise"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_view_form
msgid "Confirm selected company ?"
msgstr "Valider l'entreprise sélectionnée ?"

#. module: l10n_fr_siren_lookup
#: model:ir.model,name:l10n_fr_siren_lookup.model_res_partner
msgid "Contact"
msgstr ""

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__country_id
msgid "Country"
msgstr "Pays"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__create_uid
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__create_uid
msgid "Created by"
msgstr "Créé par"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__create_date
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__create_date
msgid "Created on"
msgstr "Créé le"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__creation_date
msgid "Creation date"
msgstr "Date de création"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_res_partner__display_name
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__display_name
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__display_name
msgid "Display Name"
msgstr "Nom affiché"

#. module: l10n_fr_siren_lookup
#: code:addons/l10n_fr_siren_lookup/models/res_partner.py:0
#, python-format
msgid ""
"Failed to query VIES.\n"
"Technical error: %s."
msgstr ""
"Erreur pendant connexion au service VIES.\n"
"Erreur technique : %s."

#. module: l10n_fr_siren_lookup
#: code:addons/l10n_fr_siren_lookup/models/res_partner.py:0
#, python-format
msgid ""
"Failure in the request on data.opendatasoft.com to create or update partner "
"from SIREN or SIRET. Technical error: %s."
msgstr ""
"Echec de la requête auprès de data.opendatasoft.com pour créer ou mettre à jour le contact "
"à partir du SIREN ou SIRET. Erreur technique : %s."

#. module: l10n_fr_siren_lookup
#: model:ir.model,name:l10n_fr_siren_lookup.model_siren_wizard
msgid "Get values from companies"
msgstr "Obtenir les informations sur les entreprises"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_res_partner__id
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__id
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__id
msgid "ID"
msgstr ""

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_res_partner____last_update
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard____last_update
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line____last_update
msgid "Last Modified on"
msgstr "Dernière modification le"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__write_uid
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__write_uid
msgid "Last Updated by"
msgstr "Dernière mise à jour par"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__write_date
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__write_date
msgid "Last Updated on"
msgstr "Dernière mise à jour le"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
msgid "Legal Infos"
msgstr "Infos Légales"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__legal_type
msgid "Legal Type"
msgstr "Forme Juridique"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_view_form
msgid "Lookup"
msgstr "Recherche"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__name
msgid "Name"
msgstr "Nom"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__name
msgid "Name to Search"
msgstr "Nom à Rechercher"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__partner_id
msgid "Partner"
msgstr "Contact"

#. module: l10n_fr_siren_lookup
#: code:addons/l10n_fr_siren_lookup/wizard/siren_wizard.py:0
#, python-format
msgid "Partner '%s' is not a company. This action is not relevant."
msgstr "Le Contact '%s' n'est pas une société. Cette action n'est appropriée."

#. module: l10n_fr_siren_lookup
#: code:addons/l10n_fr_siren_lookup/wizard/siren_wizard.py:0
#, python-format
msgid "Partner updated via the opendatasoft.com API."
msgstr "Contact mis à jour via l'API opendatasoft.com"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard__line_ids
msgid "Results"
msgstr "Résultats"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__siren
msgid "SIREN"
msgstr "SIREN"

#. module: l10n_fr_siren_lookup
#: model:ir.actions.act_window,name:l10n_fr_siren_lookup.siren_lookup_partner_action
msgid "SIREN Lookup"
msgstr "Recherche SIREN"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_view_form
msgid "SIREN Lookup via opendatasoft.com"
msgstr "Recherche SIREN via opendatasoft.com"

#. module: l10n_fr_siren_lookup
#: model:ir.actions.act_window,name:l10n_fr_siren_lookup.siren_wizard_action
msgid "SIREN lookup via opendatasoft.com"
msgstr "Informations base Siren"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__siret
msgid "SIRET"
msgstr "SIRET"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_view_form
msgid "Search Results"
msgstr "Résultats de Recherche"

#. module: l10n_fr_siren_lookup
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_line_view_form
#: model_terms:ir.ui.view,arch_db:l10n_fr_siren_lookup.siren_wizard_view_form
msgid "Select"
msgstr "Sélectionner"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__street
msgid "Street"
msgstr "Rue"

#. module: l10n_fr_siren_lookup
#: code:addons/l10n_fr_siren_lookup/models/res_partner.py:0
#, python-format
msgid "The webservice data.opendatasoft.com returned an HTTP error code %s."
msgstr "Le service data.opendatasoft.com a retourné le code d'erreur HTTP %s."

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__wizard_id
msgid "Wizard"
msgstr "Assistant"

#. module: l10n_fr_siren_lookup
#: model:ir.model.fields,field_description:l10n_fr_siren_lookup.field_siren_wizard_line__zip
msgid "Zip"
msgstr "Code Postal"

```

**Review by remi-filament — CHANGES_REQUESTED** [review]

File: l10n_fr_siren_lookup/models/res_partner.py
Comment (alexis-via) [reply l10n_fr_siren_lookup/models/res_partner.py]: arf, you should not use these old stdnum versions that don't support the new format for Northern Ireland VAT numbers. Maybe we could check the stdnum version in the code and add the argument when stdnum is 1.9+.

Comment (remi-filament) [reply l10n_fr_siren_lookup/models/res_partner.py]: Good idea to make the command dependant of stdnum version.
I guess we cannot assume that anyone using this module would update stdnum from Odoo required version (nor may care about Northern Ireland VAT :))

Comment (alexis-via) [reply l10n_fr_siren_lookup/models/res_partner.py]: Poor Northern Ireland people! Many companies prefer to stick to an old outdated stdnum version because they want to stick to what is written in requirements.txt of Odoo and, as a consequence, they can't sell to Northern Ireland any more. I guess the problem is not just with odoo... I would be interested to know how many companies refuse to sell to Northern Ireland for software reasons! No surprise their shops are empty. And, in Odoo v15, requirements.txt is still on stdnum 1.8, cf https://github.com/odoo/odoo/blob/15.0/requirements.txt#L49 This is not serious, so just don't follow requirements.txt of Odoo.

**Comment by alexis-via** [comment]
I renamed the module to l10n_fr_siret_lookup.
I now test the version number of stdnum, so it should work with stdnum 1.8

**Comment by alexis-via** [comment]
FYI, VIES is down at the moment. So it's not the right moment to test!!!

**Review by remi-filament — APPROVED** [review]
Working fine on my side
I however get an error on Runbot : Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed 

I am wondering if this could be related to the recent change in ([LetsEncrypt root certificate from 1st of October)[https://letsencrypt.org/2021/10/01/cert-chaining-help.html]

Would it be possible that runbot is running on old openssl version ?

**Comment by remi-filament** [comment]
Hi @gurneyalex on this module using connection on [OpenDataSoft external API](https://data.opendatasoft.com), Odoo runbot reports the following error : `Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed`

I have seen this error a few times since beginning of October, mainly on website with Let'sEncrypt certificate (like the API one we are trying to reach), due to change on root chain for Let'sEncrypt certificate since 1st of October this year. Old versions of OpenSSL (<= 1.0.2) will often raise this error.

Do you think this might be related ?

Usually updating openssl solves the issue, otherwise there are a few workarounds when keeping openssl v1.0.2 [here](https://www.openssl.org/blog/blog/2021/09/13/LetsEncryptRootCertExpire/)

Thanks !

**Pull request merged**
